VimeoUpload/Descriptor System/ReachableDescriptorManager.swift

```diff
@@ -93,7 +96,13 @@ import VimeoNetworking
             backgroundSessionManager = VimeoSessionManager.backgroundSessionManager(identifier: backgroundSessionIdentifier, baseUrl: VimeoBaseURL, accessTokenProvider: accessTokenProvider, apiVersion: apiVersion)
         }
         
-        super.init(sessionManager: backgroundSessionManager, name: name, archivePrefix: archivePrefix, shouldLoadArchive: shouldLoadArchive, documentsFolderURL: documentsFolderURL, delegate: descriptorManagerDelegate)
+        super.init(sessionManager: backgroundSessionManager,
+                   name: name,
+                   archivePrefix: archivePrefix,
+                   shouldLoadArchive: shouldLoadArchive,
+                   documentsFolderURL: documentsFolderURL,
+                   migrator: migrator,
+                   delegate: descriptorManagerDelegate)
```

Comment:
@nicolelehrer you are right that migration should be Vimeo iOS's business, and that other VimeoUpload users should not know about this migrator object. From what I can see, there are two ways we can execute this one:

1. Let the classes that need data migration instantiate the migrator object in their constructors. That is not a problem because in Vimeo iOS, the concrete migrator class always points to a sandbox, and in the case of share extension, it will not do anything due to the extension's sandbox being empty all the time.
2. Let Vimeo iOS migrate the files over before instantiating the VimeoUpload stack.

No. 1 is easier for me to do because all of the important pieces are already there, so the work is virtually zero. If you are okay with this plan, I'll go ahead and move things around.